include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_fr_photo_detail($type) {
+		if (api_user()===false) throw new ForbiddenException();
+		if(!x($_REQUEST,'photo_id')) throw new BadRequestException("No photo id.");
 
 		$scale = (x($_REQUEST, 'scale') ? intval($_REQUEST['scale']) : false);
-		$scale_sql = ($scale === false ? "" : sprintf("AND `scale`=%d",intval($scale)));
-		$data_sql = ($scale === false ? "" : "ANY_VALUE(`data`) AS data`,");
-
-		$r = q("SELECT %s ANY_VALUE(`resource-id`) AS `resource-id`, ANY_VALUE(`created`) AS `created`,
-				ANY_VALUE(`edited`) AS `edited`, ANY_VALUE(`title`) AS `title`, ANY_VALUE(`desc`) AS `desc`,
-				ANY_VALUE(`album`) AS `album`, ANY_VALUE(`filename`) AS `filename`, ANY_VALUE(`type`) AS `type`,
-				ANY_VALUE(`height`) AS `height`, ANY_VALUE(`width`) AS `width`, ANY_VALUE(`datasize`) AS `datasize`,
-				ANY_VALUE(`profile`) AS `profile`, min(`scale`) as minscale, max(`scale`) as maxscale
-				FROM `photo` WHERE `uid` = %d AND `resource-id` = '%s' %s",
+		$photo_id = $_REQUEST['photo_id'];
+
+		// prepare json/xml output with data from database for the requested photo
+		$data = prepare_photo_data($type, $scale, $photo_id);
+
+		return api_format_data("photo_detail", $type, $data);
+	}
+
+
+	/**
+	 * @brief updates the profile image for the user (either a specified profile or the default profile)
+	 *
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_account_update_profile_image($type) {
+		if (api_user()===false)
+			throw new ForbiddenException();
+
+		// input params
+		$profileid = (x($_REQUEST,'profile_id') ? $_REQUEST['profile_id'] : 0);
+
+		// error if image data is missing
+		if(!x($_FILES,'image')) {
+			throw new BadRequestException("no media data submitted");
+		}
+
+		// check if specified profile id is valid
+		if ($profileid != 0) {
+			$r = q("SELECT * FROM `profile` WHERE `uid` = %d AND `id` = %d",
+				intval(api_user()),
+				intval($profileid));
+			// error message if specified profile id is not in database
+			if (!dbm::is_result($r))
+				throw new BadRequestException("profile_id not available");
+			$is_default_profile = $r['profile'];
+		} else {
+			$is_default_profile = 1;
+		}
+
+		// get mediadata from image or media (Twitter call api/account/update_profile_image provides image)
+		$media = null;
+		if(x($_FILES, 'image'))
+			$media = $_FILES['image'];
+		elseif(x($_FILES, 'media'))
+			$media = $_FILES['media'];
+
+		// save new profile image
+		$data = SaveMediaToDatabase("profileimage", $media, $type, t('Profile Photos'), "", "", "", "", "", $is_default_profile);
+
+		// get filetype
+		if (is_array($media['type']))
+			$filetype = $media['type'][0];
+		else
+			$filetype = $media['type'];
+
+		if ($filetype = "image/jpeg")
+			$fileext = "jpg";
+		else if ($filetype = "image/png")
```

Comment:
I believe you confused `=` and `==`.
Standards: Please add braces to this condition.